railties/lib/rails/application.rb

```diff
@@ -404,6 +404,32 @@ def secrets=(secrets) #:nodoc:
       @secrets = secrets
     end
 
+    # The secret_key_base is used as the input secret to the application's key generator, which in turn
```

Comment:
Had this lined up to deprecate `secrets` but looks like it's a bit more involved than that. Let's defer the deprecation to after this is merged.

```ruby
ActiveSupport::Deprecation.warn(<<-end_of_message.strip_heredoc)
  `Rails.application.secrets` has been deprecated in favor of `Rails.application.credentials`
  and will be removed in Rails 6.0.

  See `bin/rails credentials --help` for more.
end_of_message
```

(Most trouble is just that our test suite relies on `secrets`, so will be seeing some deprecation warnings.)